cpukit/score/src/schedulerstrongapa.c

```diff
@@ -424,20 +939,65 @@ Thread_Control *_Scheduler_strong_APA_Remove_processor(
   );
 }
 
-void _Scheduler_strong_APA_Yield(
+void _Scheduler_strong_APA_Node_initialize(
   const Scheduler_Control *scheduler,
+  Scheduler_Node          *node,
   Thread_Control          *the_thread,
-  Scheduler_Node          *node
+  Priority_Control         priority
 )
 {
-  Scheduler_Context *context = _Scheduler_Get_context( scheduler );
+  Scheduler_SMP_Node *smp_node;
+  Scheduler_strong_APA_Node *strong_node;
 
-  _Scheduler_SMP_Yield(
-    context,
-    the_thread,
-    node,
-    _Scheduler_strong_APA_Extract_from_ready,
-    _Scheduler_strong_APA_Enqueue,
-    _Scheduler_strong_APA_Enqueue_scheduled
+  smp_node = _Scheduler_SMP_Node_downcast( node );
+  strong_node = _Scheduler_strong_APA_Node_downcast( node );
+
+  _Scheduler_SMP_Node_initialize( scheduler, smp_node, the_thread, priority );
+
+  _Processor_mask_Assign(
+    &strong_node->Affinity,
+   _SMP_Get_online_processors()
   );
```

Comment:
Here should you confirm Affinity is nonzero?

Reply:
The non-zero affinity is checked once when the node asks for help. Should I have an assert here as well?

Reply:
as long as you do it somewhere and it is well-defined behavior, then it is ok.

Reply:
Thanks :)